File: src/main/java/com/jakeesveld/zoos/controller/ZooController.java
package com.jakeesveld.zoos.controller;

import com.jakeesveld.zoos.model.Zoo;
import com.jakeesveld.zoos.service.ZooService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
public class ZooController {

    @Autowired
    private ZooService zooService;

    @GetMapping(value = "/zoos/zoos", produces = {"application/json"})
    public ResponseEntity<?> getZooDetails() {
        return new ResponseEntity<>(zooService.getZooDetails(), HttpStatus.OK);
    }

    @GetMapping(value = "/zoos/zoos/all", produces = {"application/json"})
    public ResponseEntity<?> getAllZoos(){
        return new ResponseEntity<>(zooService.findAll(), HttpStatus.OK);
    }

    @PutMapping(value = "/admin/zoos/{id}")
    public ResponseEntity<?> updateZoo(@PathVariable long id, @RequestBody Zoo updatedZoo) {
        zooService.updateZoo(id, updatedZoo);
        return new ResponseEntity<>(HttpStatus.OK);
    }

    @PostMapping(value = "/admin/zoos")
    public ResponseEntity<?> addZoo(@RequestBody Zoo zoo) {
        zooService.addZoo(zoo);
        return new ResponseEntity<>(HttpStatus.OK);
    }

    @DeleteMapping(value = "/admin/zoos/{id}")
    public ResponseEntity<?> deleteZoo(@PathVariable long id){
        zooService.deleteZoo(id);
        return new ResponseEntity<>(HttpStatus.OK);
    }
}

File: src/main/java/com/jakeesveld/zoos/controller/ZooController.kt
package com.jakeesveld.zoos.controller

import com.jakeesveld.zoos.model.Zoo
import com.jakeesveld.zoos.service.ZooService
import com.jakeesveld.zoos.view.ZooDetails
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
class ZooController {

    @Autowired
    private var zooService: ZooService? = null

    val zooDetails: ResponseEntity<*>
        @GetMapping(value = ["/zoos/zoos"], produces = ["application/json"])
        get() = ResponseEntity(zooService!!.zooDetails, HttpStatus.OK)

    val allZoos: ResponseEntity<*>
        @GetMapping(value = ["/zoos/zoos/all"], produces = ["application/json"])
        get() = ResponseEntity(zooService!!.findAll(), HttpStatus.OK)

    @PutMapping(value = ["/admin/zoos/{id}"])
    fun updateZoo(@PathVariable id: Long, @RequestBody updatedZoo: Zoo): ResponseEntity<*> {
        zooService!!.updateZoo(id, updatedZoo)
        return ResponseEntity<Any>(HttpStatus.OK)
    }

    @PostMapping(value = ["/admin/zoos"])
    fun addZoo(@RequestBody zoo: Zoo): ResponseEntity<*> {
        zooService!!.addZoo(zoo)
        return ResponseEntity<Any>(HttpStatus.OK)
    }

    @DeleteMapping(value = ["/admin/zoos/{id}"])
    fun deleteZoo(@PathVariable id: Long): ResponseEntity<*> {
        zooService!!.deleteZoo(id)
        return ResponseEntity<Any>(HttpStatus.OK)
    }
}
